designs/administrator/future-openshift/bulk-actions/index.md

```diff
@@ -0,0 +1,139 @@
+---
+parent: Administrator
+version: 4.x
+---
+
+# Bulk actions
+
+Bulk actions will be added to a number of list views in the OpenShift console. This feature will allow users to perform an action on multiple resources at once. The list of actions include adding labels, adding annotations and deleting resources.
+
+## Bulk actions in toolbar
+The bulk selector will be the first component in the toolbar on pages where bulk actions are available.
+![toolbar](img/toolbar.png)
+
+Users can click the checkbox in the bulk selector to select all items in the list. If all items are selected, the bulk selector checkbox will be filled with a check and the bulk selector will read ‘[#] selected’, informing users that all known items in the list are selected. Clicking the bulk selector checkbox again will deselect all items.
+![bulk-selector-states](img/bulk-selector-states.png)
+
+Users can also choose to select a group of items by clicking the associated checkboxes next to them. The bulk selector checkbox will be filled in with a ‘-‘ to indicate that some items are selected, and the bulk selector will display the number of items selected (e.g. ‘2 selected’). Clicking the bulk selector checkbox will deselect all items.
+![add-labels](img/add-labels.png)
+
+## Behavior
+If no items are selected, the bulk actions dropdown will be enabled, shown as a basic dropdown, but the actions within will be disabled. A tooltip will explain to users that they must select resources in order to take actions.
+![none-checked](img/none-checked.png)
+
+When items are selected, the bulk actions basic dropdown will turn into a primary dropdown, and actions within it will be enabled. 
```

Comment:
I feel like it was a thing to not have 2 primary buttons at some point? Just figured I'd mention.

Reply:
@itsptk yes that's where this came in actually! We decided to only activate that dropdown as primary once items were selected, because at that point you'd imagine the user is looking to take action on those items so the dropdown becoming enabled makes sense. We did end up coming up with an animated version of this... not sure how possible that is though https://xd.adobe.com/view/4e189954-36b5-4db0-4d5b-278b452fd6fc-3829/